pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
+#
+#  histogram_aggregates:
+#    - max
+#    - median
+#    - avg
+#    - count
+
+## @param histogram_percentiles - list of strings - optional - default: "0.95"
+## Configure which percentiles are computed by the Agent. It must be a list of float between 0 and 1.
+## Warning: percentiles must be specified as yaml strings
 #
-#    tag_value_split_separator:
-#      foo: ;
+#  histogram_percentiles:
+#    - "0.95"
+
+## @param histogram_copy_to_distribution - boolean - optional - default: false
+## Copy histogram values to distributions for true global distributions (in beta)
+## Note: This increases the number of custom metrics created.
 #
-#  will result in the raw tag being transformed into "foo:1", "foo:2", "foo:3" tags
+#  histogram_copy_to_distribution: false
 
-# Configure the level of granularity of tags to send for checks and dogstatsd metrics and events.
-# Choices are:
-# - low: add tags about low-cardinality objects (clusters, hosts, deployments, container images, ...)
-# - orchestrator: add tags about pod, (in Kubernetes), or task (in ECS or Mesos) -level of cardinality
-# - high: add tags about high-cardinality objects (individual containers, user IDs in requests, ...)
-# WARNING: sending container tags for dogstatsd metrics may create more metrics
-# (one per container instead of one per host). This may impact your custom metrics billing.
+## @param histogram_copy_to_distribution_prefix - string - optional
+## A prefix to add to distribution metrics created when histogram_copy_to_distributions is true
 #
-# checks_tag_cardinality: orchestrator
-# dogstatsd_tag_cardinality: low
+#  histogram_copy_to_distribution_prefix: "<PREFIX>"
 
-# Histogram and Historate configuration
+## @parmam forwarder_timeout - integer - optional - default: 20
+## Forwarder timeout in seconds
 #
-# Configure which aggregated value to compute. Possible values are: min, max,
-# median, avg, sum and count.
+#  forwarder_timeout: 20
+
+## @param forwarder_retry_queue_max_size - integer - optional - default: 30
+## The forwarder retries failed requests. Use this setting to change the
+## maximum length of the forwarder's retry queue (each request in the queue
+## takes no more than 2MB in memory)
+#
+#  forwarder_retry_queue_max_size: 30
+
+## @param forwarder_num_workers - integer - optional - default: 1
+## The number of workers used by the forwarder. Note that each worker opens an outbound HTTP
+## connection towards Datadog's metrics intake at every flush.
+#
+#  forwarder_num_workers: 1
+
+## @param collect_ec2_tags - boolean - optional - default: false
+## Collect AWS EC2 custom tags as Agent tags.
+#
+#  collect_ec2_tags: false
+
+## @param collect_gce_tags - boolean - optional - default: true
+## Collect Google Cloud Engine metadata as Agent tags
 #
-# histogram_aggregates: ["max", "median", "avg", "count"]
+#  collect_gce_tags: true
+
+{{ end }}
+{{- if .Agent }}
+
+###############################
+## Agent Extra Configuration ##
```

Comment:
Do we really want to have the agent extra configuration before other collectors?